transport/src/main/java/io/netty/channel/pool/SimpleChannelPool.java

```diff
@@ -389,13 +393,47 @@ protected boolean offerChannel(Channel channel) {
 
     @Override
     public void close() {
+        closeIdleChannels(nextEventLoop());
+    }
+
+    void closeIdleChannels(EventExecutor executor) {
+        Promise<Void> result = executor.newPromise();
+        closeIdleChannels(result);
+        if (!executor.inEventLoop()) {
+            result.awaitUninterruptibly();
+        }
+    }
+
+    private void closeIdleChannels(final Promise<Void> result) {
+        final Set<Channel> channelsToClose = Collections.newSetFromMap(new ConcurrentHashMap<Channel, Boolean>());
         for (;;) {
             Channel channel = pollChannel();
             if (channel == null) {
                 break;
             }
-            // Just ignore any errors that are reported back from close().
-            channel.close().awaitUninterruptibly();
+            channelsToClose.add(channel);
+        }
+        if (channelsToClose.isEmpty()) {
+            // no idle channels in the pool - nothing to close
+            result.setSuccess(null);
```

Comment:
`trySuccess(...)` ?

Reply:
@normanmaurer, why was it suggested to swap to trySuccess here?

Reply:
@ejona86 couldn't this race ?

Reply:
Doesn't look like it. It is a private method that is only used in one spot. The passed in promise is not used before this method and was not leaked to any other objects.